### PR TITLE
Add rich editor strings

### PR DESCRIPTION
**Pull request by charrondev**
Closes https://github.com/vanilla/vanilla/issues/7454
~~~blocked by https://github.com/vanilla/rich-editor/pull/110~~~

<!---
Thanks for filing an issue 😄 ! Before you submit, please read the following:

Translations in this repository are managed by Transifex https://transifex.com/vanilla/vanilla. To contribute translations, please create an account on Transifex.com and request access to a team via the Vanilla Forums Community Forum https://vanillaforums.org/discussions. We'd love your help!

To add a new source string in `tx-source` please proceed with the template below.
-->

<!--- Please explain below where in the Vanilla Forums product or repos this translation string will be used.

If the string is not present in one of Vanilla's github repos it likely doesn't belong in this repo. Instead it can be added directly in your addon! https://docs.vanillaforums.com/developer/locales/#overriding-locales
-->

## New or changed source translation strings

This PR adds all translations strings from the Rich Editor.

| Source String                | Location or PR in a Vanilla repository
|------------------------------|----------------------------------------
|            All of the added source strings                  | https://github.com/vanilla/rich-editor|

### CODE REVIEW

File: tx-source/site_core.php [hunk 1]
```diff
@@ -1696,12 +1696,77 @@
...
+$Definition['Activities'] = 'Activities';
+$Definition['Objects'] = 'Objects';
+$Definition['Symbols'] = 'Symbols';
+$Definition['Flags'] = 'Flags';
+
+// Media Insertion
+$Definition['Image'] = 'Image';
+$Definition['Insert Media'] = 'Insert Media';
+$Definition['Paste the URL of the media you want.'] = 'Paste the URL of the media you want.';
+$Definition['http://'] = 'http://';
```
[the remark this answers is not in the document]
Comment (charrondev) [reply tx-source/site_core.php]: I've removed them from here, but they are still translation strings in the core. This is mostly for consistency with out linting rule that every appearing in the UI must go through a translation function.

**Comment by charrondev** [comment]
Spoiler warning has been added here.

File: tx-source/site_core.php [hunk 2]
```diff
@@ -1696,12 +1696,75 @@
...
-$Definition['food-drink'] = 'Food and Drink';
-$Definition['travel-places'] = 'Travel and Places';
-$Definition['activities'] = 'Activities';
-$Definition['objects'] = 'Objects';
-$Definition['symbols'] = 'Symbols';
-$Definition['flags'] = 'flags';
+// Rich Editor
+// Dashboard
+$Definition['Enable Rich Quotes'] = 'Enable Rich Quotes';
+$Definition['RichEditor.QuoteEnable.Notes'] = 'Use the following option to enable quotes for the Rich Editor. This will only apply if the default formatter is "Rich".';
```
Comment (tburry) [inline tx-source/site_core.php]: These two need to go into dash_core.php.

File: tx-source/site_core.php [hunk 3]
```diff
@@ -1696,12 +1696,75 @@
...
+$Definition['Format as code block'] = 'Format as code block';
+$Definition['Spoiler'] = 'Spoiler';
+$Definition['Format as spoiler'] = 'Format as spoiler';
+$Definition['Inline Menu Available'] = 'Inline Menu Available';
+
+// Emoji Picker
+$Definition['Emoji'] = 'Emoji';
+$Definition['Emojis'] = 'Emojis';
+$Definition['Insert an emoji in your message.'] = 'Insert an emoji in your message.';
+$Definition['Use keyboard shortcuts "page up" and "page down" to cycle through available categories when menu is open.'] = 'Use keyboard shortcuts "page up" and "page down" to cycle through available categories when menu is open.';
```
Comment (tburry) [inline tx-source/site_core.php]: The key for this should be much shorter. Something like `'Use page up and page down to jump to categories.'` The translation can be the same.

Comment (charrondev) [reply tx-source/site_core.php]: I feel this confuses things a bit more. A short string key like `richeditor.emojiPicker.pagingInstructions` would be more appropriate if we were to shorten the key.

Comment (charrondev) [reply tx-source/site_core.php]: https://github.com/vanilla/vanilla/pull/7700

**Pull request merged**